### PR TITLE
Replace boolean isSpecType with enum

### PR DESCRIPTION
**Pull request by matthchr**
Just some small code cleanup

### CODE REVIEW

File: hack/generator/pkg/astmodel/armconversion/shared.go [hunk 1]
```diff
@@ -31,9 +31,9 @@ type conversionBuilder struct {
 type TypeKind int
 
 const (
-	OrdinaryType TypeKind = iota
-	SpecType
-	StatusType
+	TypeKindOrdinary TypeKind = iota
+	TypeKindSpec
+	TypeKindStatus
```
Comment (theunrepentantgeek) [inline hack/generator/pkg/astmodel/armconversion/shared.go:36]: Suggest using the type name as a suffix - `OrdinaryTypeKind`, `SpecTypeKind`, and `StatusTypeKind` - as that's easier to read. 

If you limited the scope of the enum by making it internal, you could even get away with `ordinary`, `spec` and `status` as the values.

Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/armconversion/shared.go:36]: Oh, I see elsewhere that they're used outside the package - but my first suggestion is still useful.

**Review by theunrepentantgeek — APPROVED** [review]
Looks good - one minor comment about naming.

gif

File: hack/generator/pkg/astmodel/armconversion/shared.go (hunk 1 above)
Comment (matthchr) [reply hack/generator/pkg/astmodel/armconversion/shared.go:36]: We don't always put the enum type in the enum value name, but when we do it's as a prefix as far as I can tell. See `ValidationKind`, `GenerationPipeline`, `ExportFilter`, `TypeFilter`, etc.

There also seems to be at least some of this standard in the community as well, seeing as the [proposal for Go2](https://github.com/golang/go/issues/19814) also uses prefixes (in its example).

I wasn't able to find anywhere we used a suffix (there are places where we don't prefix at all). For now I think I'm going to leave this as is. We can maybe have an overall style discussion on what we want to do here (I'm happy to change this I just want it to match what we do elsewhere too).

**Pull request merged**